src/admin/resolvers/mutations/ApprovedItem.ts

```diff
@@ -245,3 +258,105 @@ export async function uploadApprovedItemImage(
   const image = await data;
   return await uploadImageToS3(context.s3, image);
 }
+
+/**
+ * Imports an approved item
+ * Creates an approved item if it doesn't exist, and
+ * creates a scheduled item for the approved item
+ * if it doesn't exist
+ * @param parent
+ * @param data
+ * @param context
+ */
+export async function importApprovedItem(
+  parent,
+  { data },
+  context: IContext
+): Promise<ImportApprovedCuratedCorpusItemPayload> {
+  // check if user is authorized to import an item
+  if (!context.authenticatedUser.canWriteToCorpus()) {
+    throw new AuthenticationError(ACCESS_DENIED_ERROR);
+  }
+
+  let approvedItem = await getApprovedItemByUrl(context.db, data.url);
+  if (!approvedItem) {
+    approvedItem = await dbImportApprovedItem(
```

Comment:
we should also be calling the `checkLanguage()` function that's called in ApprovedItem mutation: https://github.com/Pocket/curated-corpus-api/blob/main/src/admin/resolvers/mutations/ApprovedItem.ts#L75

Reply:
so we should be throwing an error if the language is not valid, correct?

Reply:
done